packetbeat/protos/tls/parse.go

```diff
@@ -488,10 +491,40 @@ func (version tlsVersion) String() string {
 	return fmt.Sprintf("%d.%d", version.major, version.minor)
 }
 
+func getKeySize(key interface{}) int {
+	if key == nil {
+		return 0
+	}
+	switch pubKey := key.(type) {
+	case *rsa.PublicKey:
+		if n := pubKey.N; n != nil {
+			return n.BitLen()
+		}
+
+	case *dsa.PublicKey:
+		if p := pubKey.Parameters.P; p != nil {
+			return p.BitLen()
+		}
+		if y := pubKey.Y; y != nil {
+			return y.BitLen()
+		}
+
+	case *ecdsa.PublicKey:
+		if params := pubKey.Params(); params != nil {
+			return params.BitSize
+		}
+		if y := pubKey.Y; y != nil {
+			return y.BitLen()
+		}
+	}
+	return 0
+}
+
 func certToMap(cert *x509.Certificate, includeRaw bool) common.MapStr {
 	certMap := common.MapStr{
 		"signature_algorithm":  cert.SignatureAlgorithm.String(),
 		"public_key_algorithm": toString(cert.PublicKeyAlgorithm),
+		"public_key_size":      getKeySize(cert.PublicKey),
```

Comment:
WDYT about not adding the field if we can't determine the value? I'm wondering if that would make it easier to use.

Reply:
Yes that makes more sense.